Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40432&attempt_id=d3cba387-1099-40bb-bef8-ddfd75d26f12)

Attempted: nasm

<details><summary>Partial log (click to expand)</summary><p>

```
gcc -c -g -O3 -fwrapv -U__STRICT_ANSI__ -fno-common -Werror=attributes -W -Wall -pedantic -Wc90-c99-compat -Wno-long-long -Werror=implicit -Werror=missing-braces -Werror=return-type -Werror=trigraphs -Werror=pointer-arith -Werror=missing-prototypes -Werror=missing-declarations -Werror=comment -Werror=vla -DHAVE_CONFIG_H -I. -I. -I./include -I./include -I./x86 -I./x86 -I./asm -I./asm -I./disasm -I./disasm -I./output -I./output -o asm/nasm.o asm/nasm.c
In file included from ./include/nasm.h:46,
                 from asm/nasm.c:47:
./include/nasmlib.h:194:1: error: 'pure' attribute on function returning 'void' [-Werror=attributes]
 void pure_func seg_init(void);
 ^~~~
cc1: some warnings being treated as errors
make: *** [Makefile:75: asm/nasm.o] Error 1
builder for '/nix/store/q1ll3km6l5crfz1sjb8ixfpwd8v5iq6m-nasm-2.13.03.drv' failed with exit code 2
error: build of '/nix/store/q1ll3km6l5crfz1sjb8ixfpwd8v5iq6m-nasm-2.13.03.drv' failed
```
</p></details>

--- a/pkgs/development/libraries/libglvnd/default.nix
+++ b/pkgs/development/libraries/libglvnd/default.nix
@@ -20,6 +20,8 @@ in stdenv.mkDerivation rec {
     "-UDEFAULT_EGL_VENDOR_CONFIG_DIRS"
     # FHS paths are added so that non-NixOS applications can find vendor files.
     "-DDEFAULT_EGL_VENDOR_CONFIG_DIRS=\"${driverLink}/share/glvnd/egl_vendor.d:/etc/glvnd/egl_vendor.d:/usr/share/glvnd/egl_vendor.d\""
+
+    "-Wno-error=array-bounds"
   ];
 
   # Indirectly: https://bugs.freedesktop.org/show_bug.cgi?id=35268

